src/utils/with-react.js

```diff
@@ -73,4 +79,61 @@ const applyDefaults = ceInst => {
 
 export { applyDefaults };
 
-export default createElement => val(createElement);
+const distributeProperties = ({ children, ...properties }, componentClass) => {
+  // initialize
+  const attrs = {};
+  const props = {};
+  let map;
+  // iterate over all properties
+  Object.keys(properties).forEach(name => {
+    let value = properties[name];
+    // classify property by type to select correct map object
+    // (note that unregistered properties are classified as attr(ibute)s via their undefined .type)
+    const event = name.indexOf('on') === 0 && Function; // Function if onXXX event name, false otherwise
+    const className = name === 'className' && name;
+    const defaultType = { type: event };
+    const declaredType = (componentClass.properties[name] || defaultType).type;
+    const type = event || className || declaredType;
```

Comment:
Spontan würde man alle 3 Werte gleichberechtigt sehen. Was sie aber nicht sind. Sehe das auch als gefährlich zum Warten. Evtl. einfach die ( ) Klammern dort setzen wo es auch die Engine machen würde.

Reply:
done via if's, satisfies the concern in another way